### PR TITLE
vmalert: fix possible template overwritten between rule annotations

### PR DESCRIPTION
**Pull request by Haleygo**
Reusable template can only be defined via `-rule.templates` flag, template defined inside rule annotation shouldn't be reused by others.

### CODE REVIEW

File: app/vmalert/notifier/alert.go [hunk 1]
```diff
@@ -146,12 +146,21 @@ func templateAnnotations(annotations map[string]string, data AlertTplData, tmpl
...
 		buf.Reset()
 		builder.Reset()
 		builder.Grow(len(header) + len(text))
 		builder.WriteString(header)
 		builder.WriteString(text)
-		if err := templateAnnotation(&buf, builder.String(), tData, tmpl, execute); err != nil {
+		// clone a new template for each parse to avoid collision
+		ctmpl, _ := tmpl.Clone()
+		ctmpl = ctmpl.Option("missingkey=zero")
+		if err := templateAnnotation(&buf, builder.String(), tData, ctmpl, execute); err != nil {
```
Comment (f41gh7) [inline app/vmalert/notifier/alert.go:163]: WDYT of making `tmpl.Clone` a part of `templateAnnotation` function? 

Also, I think it's better to handle `error` from `tmpl.Clone()` call. Currently, it cannot return `err`. But later this API can be changed and it'd be great to handle errors consistently.

**Review by f41gh7 — APPROVED** [review]
LGTM

File: app/vmalert/notifier/alert.go (hunk 1 above)
Comment (Haleygo) [reply app/vmalert/notifier/alert.go:163]: will have a follow up pull request to refactor template function calls, but want to separate bug fixes first.

**Comment by f41gh7** [comment]
Thanks for contribution!

**Pull request merged**